example/test/function_component_test.dart

```diff
@@ -183,6 +183,53 @@ UseRefTestComponent(Map props) {
   ]);
 }
 
+int fibonacci(int n) {
+  if (n <= 1) {
+    return 1;
+  }
+  return fibonacci(n - 1) + fibonacci(n - 2);
+}
+
+final useMemoTestFunctionComponent = react.registerFunctionComponent(UseMemoTestComponent, displayName: 'useMemoTest');
+
+UseMemoTestComponent(Map props) {
+  final reRender = useState(0);
+  final num = useState(35);
```

Comment:
#nit probably shouldn't use `num` as a variable name - since its a reserved word / primitive type in Dart's type system.